City-of-Helsinki/drupal-helfi-platform-config
# UHF-5894 POC taxonomy autocomplete

# Taxonomy term autocomplete [UHF-5894](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5894)
- News category autocomplete fetched from etusivu-instance

## What was done
- Fetch news categories from etusivu -instance
- Show fetched categories as autocomplete suggestions in select2 field when user starts writing term name.

## How to install
* Use Liikenne (Kymp) environment if possible.
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-5894_taxonomy_autocomplete`
* Copy external entity configurations from module install forlder to conf/cmi
* Run `make drush-updb drush-cr`

## How to test
* Add new entity reference-field to news_list paragraph, for example field_news_groups `/admin/structure/paragraphs_type/news_list/fields/add-field`
  * Select one of the external entities as the reference type: helfi: news groups, helfi: news neighbourhoods or helfi: news tags, set cardinality to unlimited, save the field
  * Update the field's widget to select2 with autocomplete
* Create new landing page
* Add news "list paragraph" to the page
  * Start writing to the newly created select2 field
  * Wait for the autocomplete response
  * You should see the autocomplete suggestions
